### DISCUSSION TITLE
Why not use test_ mask when predicting

### DISCUSSION THREAD

**Discussion by Chrisable** [pyg-team/pytorch_geometric#4619] [General]
Why not use test_ mask when predicting？What are the benefits of using all the data ?
![image](https://user-images.githubusercontent.com/49330390/167844516-8d45d522-9c67-4e40-9d27-530c6728dfc7.png)

**Comment by rusty1s** [comment]
`model.inference` applies layer-wise computation of node features to prevent the neighbor explosion problem. That is, we compute all node embeddings of layer 1 before we compute node embeddings of layer 2. As such, we need to compute the node embeddings here for all nodes (applying the `test_mask` in a later step). If you are interested in getting the predictions for all nodes, layer-wise computation is probably the best choice to do that. If you are only interested in getting the predictions for a small set of nodes, it is not necessary to do so.

**Reply by Chrisable** [reply]
You means that if i am only interested in getting the predictions for the nodes of the test_mask,i can only use the nodes of test_mask instead of all data.If so,will the accuracy drop?I worry that if we use the nodes of test_mask, when the nodes of test_mask's neighbors are not all trained, the accuracy of aggregating its neighbors will decline during the prediction period

**Reply by rusty1s** [reply]
If you are only using `test_mask` nodes during inference, the model performance will likely decrease as you have access to a smaller amount of data. Note that this does not happen here since we compute the node embeddings for all nodes in all layers, not just the ones for test nodes.
